app/views/users/index.html.erb

```diff
@@ -1,2 +1,14 @@
-<h1>User Index page</h1>
-<p> this is the user index page </p>
\ No newline at end of file
+<h1>1. User Index page</h1>
+<p> this is the user index page </p>
+
+<%# display all the users in the database %>
+<% @users.each do |user| %>
+  <div>
+    <%= image_tag user.photo%>
+    <%= link_to user_path(user.id) do %>
+        <button><%= user.name %> Posts: <%= user.posts_counter%></button>
+    <% end %>
+  </div>
+<% end %>
```

Comment:
- [Optional] You can improve on this by making use of partial templates. Instead of using @objects.each, you can create a template like this: 

```ruby
# app/views/users/_users.html.erb

<h1>1. User Index page</h1>
<p> this is the user index page </p>
<% @users.each do |user| %>
  <div>
    <%= image_tag user.photo%>
    <%= link_to user_path(user.id) do %>
        <button><%= user.name %> Posts: <%= user.posts_counter%></button>
    <% end %>
  </div>
<% end %>

```
Then use the render method within the view: 

 
```.ruby
# app/views/users/index.html.erb

    <%= render users %>

```

Partials are also useful when you need to reuse exactly the same code (DRY philosophy) :+1: